lib/src/icon/Icon.tsx

```diff
@@ -0,0 +1,38 @@
+import React from "react";
+import styled from "styled-components";
+import IconPropsType from "./types";
+
+const DxcIcon = ({ icon }: IconPropsType): JSX.Element => {
+  let filled = false;
+  let iconName = icon;
+  if (icon.startsWith("filled_")) {
+    filled = true;
+    iconName = icon.replace(/filled_/g, "");
+  }
+  return <IconContainer role="img" aria-label={icon} filled={filled} icon={iconName} />;
```

Comment:
In terms of accessibility, this is not accurate, as the role would depend on the purpose of the icon.
I would consider those scenarios:
- If the icon is **purely decorative** and does not convey any meaningful information, it can have a `role="presentation"` and `aria-hidden="true"` to indicate screen readers that it should be ignored.
- If the icon is clickable and **performs an action when activated**, it can have a `role="button"`.
- If the icon acts as a link to **navigate to another page or resource**, it can have a `role="link"`.
- Only if the icon is **purely informative** and provides important context or content to the user, it can have a `role="img"`.

Taking this into account, maybe it would be interesting to add a parameter in a similar way as it was done for the `Divider` component to decide what is the most appropiate role (being default to decorative, which will probably be the most common scenario).

Reply:
Based on WAI ARIA role should always be "img" in the case of a font representing an icon.
I agree about `aria-hidden=true `, which we can consider being the case for all of our use cases.

See [WAI ARIA doc](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA24).

Reply:
Great info!